src/fiber/channel.rs

```diff
@@ -840,6 +829,28 @@ where
         .await;
     }
 
+    // Try to settle down a held TLC (i.e., a TLC whose preimage is not available when it is received).
+    // This is usually a TLC associated with a hold invoice. We call of this function should ensure that
+    // this TLC is already in a state that can be settled down (i.e. the invoice associated with it is
+    // in a Received state and we have saved its preimage to the store).
+    async fn try_to_settle_down_held_tlc(
+        &self,
+        myself: &ActorRef<ChannelActorMessage>,
+        state: &mut ChannelActorState,
+        hash: Hash256,
+    ) {
+        let tlc_id = match state.get_received_tlc_with_hash(hash) {
+            Some(tlc) => tlc.tlc_id,
+            None => {
+                return;
+            }
+        };
+        // Only settle down this TLC if it is not already settled down.
+        if state.tlc_state.applied_add_tlcs.insert(tlc_id) {
+            self.try_to_settle_down_tlc(myself, state, tlc_id).await;
+        }
```

Comment:
```suggestion
        if let Some(tlc) = match state.get_received_tlc_with_hash(hash) {
        let tlc_id = tlc.tlc_id;
        if state.tlc_state.applied_add_tlcs.insert(tlc_id) {
            self.try_to_settle_down_tlc(myself, state, tlc_id).await;
        }
        }
```

Reply:
Done in https://github.com/nervosnetwork/fiber/pull/511/commits/dac7a4185a082d2c5ef59cd6ad4b00ae47f9bc72